mesonbuild/backend/backends.py

```diff
@@ -915,10 +907,15 @@ def _determine_ext_objs(self, extobj: 'build.ExtractedObjects', proj_dir_to_buil
                 if comp.language in LANGS_CANT_UNITY:
                     sources += srcs
                     continue
-                for i in range((len(srcs) + unity_size - 1) // unity_size):
+                if unity_size == -1:
```

Comment:
To be more DRY, you could do something like:

```
indexes = range(...) if unity_size != -1 else [0]
for i in indexes:
    ...
```

Reply:
I like that, I'll fix that in V2 along with the mypy error